src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,194 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Parses and executes commands for the Duke application.
+ */
+public class Parser {
+
+    /**
+     * Parses and executes a given command.
+     *
+     * @param input   The input command to parse.
+     * @param tasks   The list of tasks.
+     * @param storage The storage for tasks.
+     * @return The response message after executing the command.
+     * @throws DukeException If the command is invalid or execution fails.
+     */
+    public static String parseAndExecute(String input, TaskList tasks, Storage storage) throws DukeException {
+        Ui ui = new Ui();
+        if (input.equals("bye")) {
+            return ui.showGoodbyeMessage();
+        } else if (input.equals("list")) {
+            return ui.showTaskList(tasks);
+        }
+
+        String[] parts = input.split(" ", 2);
+        String commandWord = parts[0];
+        switch (commandWord) {
+            case "todo":
+                return addTodo(parts[1], tasks, ui, storage);
```

Comment:
making method calls to helper functions instead of having a big block of logic makes the code much more readable!